src/tools/fuzzing.h

```diff
@@ -2696,34 +2743,48 @@ class TranslateToFuzzReader {
         .add(FeatureSet::ReferenceTypes | FeatureSet::ExceptionHandling,
              Type::exnref));
   }
-  Type getReachableType() { return pick(getReachableTypes()); }
 
-  std::vector<Type> getConcreteTypes() {
-    return items(
-      FeatureOptions<Type>()
-        .add(FeatureSet::MVP, Type::i32, Type::i64, Type::f32, Type::f64)
-        .add(FeatureSet::SIMD, Type::v128)
-        .add(FeatureSet::ReferenceTypes,
-             Type::funcref,
-             Type::anyref,
-             Type::nullref)
-        .add(FeatureSet::ReferenceTypes | FeatureSet::ExceptionHandling,
-             Type::exnref));
+  Type getSingleConcreteType() { return pick(getSingleConcreteTypes()); }
+
+  Type getTupleType() {
+    std::vector<Type> elements;
+    size_t numElements = 2 + upTo(MAX_TUPLE_SIZE - 1);
+    elements.resize(numElements);
+    for (size_t i = 0; i < numElements; ++i) {
+      elements[i] = getSingleConcreteType();
+    }
+    return Type(elements);
   }
-  Type getConcreteType() { return pick(getConcreteTypes()); }
 
-  // Get types that can be stored in memory
-  std::vector<Type> getStorableTypes() {
-    return items(
+  Type getConcreteType() {
+    if (oneIn(3)) {
```

Comment:
Can be even lower, I think. But no scientific way to reason about how low I guess... I would say around 5-10 personally (after initial fuzzing).